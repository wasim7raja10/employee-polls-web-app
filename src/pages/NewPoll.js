import { useState } from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../actions/polls";

const NewPoll = ({ dispatch }) => {
  const navigate = useNavigate()
  const [options, setOptions] = useState({
    firstOption: '',
    secondOption: '',
  });
  const handleSubmit = (e) => {
    e.preventDefault();
    if (options.firstOption === '' || options.secondOption === '') {
      alert("Please enter both options");
      return;
    }
    dispatch(handleAddQuestion(options.firstOption, options.secondOption));
    navigate("/")
  }
  return (
    <div>
      <h1 className=" text-4xl text-center">New Poll</h1>
      <form onSubmit={handleSubmit} className="">
        <div className=" flex flex-col">
          <label data-testid="firstOptionLabel" className=" text-xl py-4" htmlFor="firstOption">First Option</label>
          <input
            value={options.firstOption}
            onChange={(e) => setOptions({ ...options, firstOption: e.target.value })}
            type="text"
            name="firstOption"
            id="firstOption"
            data-testid="firstOption"
            className="input w-full border-2 border-slate-500"
          />
        </div>

        <div className=" flex flex-col">
          <label data-testid="secondOptionLabel" className=" text-xl py-4" htmlFor="secondOption">Second Option</label>
          <input
            value={options.secondOption}
            onChange={(e) => setOptions({ ...options, secondOption: e.target.value })}
            type="text"
            name="secondOption"
            id="secondOption"
            data-testid="secondOption"
            className="input w-full border-2 border-slate-500" />
        </div>

        <button type="submit"
          data-testid="submit-poll"
          className="btn mt-6">
          Add Poll
        </button>
      </form>
    </div>
  )
}

export default connect()(NewPoll);
