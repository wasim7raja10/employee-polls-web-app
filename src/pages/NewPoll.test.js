import {fireEvent, render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import NewPoll from "./NewPoll";
import { configureStore } from "@reduxjs/toolkit";
import reducer from "../reducers";

const store = configureStore({reducer})

describe("NewPoll", () => {
    test("should render the component", () => {
        const view = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewPoll/>
                </BrowserRouter>
            </Provider>
        );
        expect(view).toBeDefined();
        expect(view).toMatchSnapshot();
    });

    test("should display all elements", () => {
        render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewPoll/>
                </BrowserRouter>
            </Provider>
        );

        const firstOptionLabelElement = screen.getByTestId("firstOptionLabel");
        const firstOptionInputElement = screen.getByTestId("firstOption");
        const secondOptionLabelElement = screen.getByTestId("secondOptionLabel");
        const secondOptionInputElement = screen.getByTestId("secondOption");
        const submitButtonElement = screen.getByTestId("submit-poll");

        expect(firstOptionLabelElement.textContent).toBe("First Option");
        expect(secondOptionLabelElement.textContent).toBe("Second Option");
        expect(submitButtonElement.textContent).toBe("Add Poll");

        fireEvent.change(firstOptionInputElement, {target: {value: 'Texas'}});
        fireEvent.change(secondOptionInputElement, {target: {value: 'New Hampshire'}});
        expect(firstOptionInputElement.value).toBe("Texas");
        expect(secondOptionInputElement.value).toBe("New Hampshire");
    });
});
