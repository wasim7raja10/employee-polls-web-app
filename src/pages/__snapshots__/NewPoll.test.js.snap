// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class=" text-4xl text-center"
        >
          New Poll
        </h1>
        <form
          class=""
        >
          <div
            class=" flex flex-col"
          >
            <label
              class=" text-xl py-4"
              data-testid="firstOptionLabel"
              for="firstOption"
            >
              First Option
            </label>
            <input
              class="input w-full border-2 border-slate-500"
              data-testid="firstOption"
              id="firstOption"
              name="firstOption"
              type="text"
              value=""
            />
          </div>
          <div
            class=" flex flex-col"
          >
            <label
              class=" text-xl py-4"
              data-testid="secondOptionLabel"
              for="secondOption"
            >
              Second Option
            </label>
            <input
              class="input w-full border-2 border-slate-500"
              data-testid="secondOption"
              id="secondOption"
              name="secondOption"
              type="text"
              value=""
            />
          </div>
          <button
            class="btn mt-6"
            data-testid="submit-poll"
            type="submit"
          >
            Add Poll
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class=" text-4xl text-center"
      >
        New Poll
      </h1>
      <form
        class=""
      >
        <div
          class=" flex flex-col"
        >
          <label
            class=" text-xl py-4"
            data-testid="firstOptionLabel"
            for="firstOption"
          >
            First Option
          </label>
          <input
            class="input w-full border-2 border-slate-500"
            data-testid="firstOption"
            id="firstOption"
            name="firstOption"
            type="text"
            value=""
          />
        </div>
        <div
          class=" flex flex-col"
        >
          <label
            class=" text-xl py-4"
            data-testid="secondOptionLabel"
            for="secondOption"
          >
            Second Option
          </label>
          <input
            class="input w-full border-2 border-slate-500"
            data-testid="secondOption"
            id="secondOption"
            name="secondOption"
            type="text"
            value=""
          />
        </div>
        <button
          class="btn mt-6"
          data-testid="submit-poll"
          type="submit"
        >
          Add Poll
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
